src/arch/xtensa/include/arch/atomic.h

```diff
@@ -65,6 +105,39 @@ static inline int32_t arch_atomic_sub(atomic_t *a, int32_t value)
 	return current;
 }
 
+#else
+
+#if CONFIG_CORE_COUNT > 1 && __XCC__
+
+#error No atomic ISA for SMP configuration
+
+#endif
+
+/* integer arithmetic methods */
+static inline int32_t arch_atomic_add(atomic_t *a, int32_t value)
+{
+	int32_t result;
+	int32_t current;
+
+	current = arch_atomic_read(a);
+	result = current - value;
+	arch_atomic_set(a, value);
+	return result;
```

Comment:
This should return current value, this way the caller know the value at the addition and the result (in a*).